addons/website_slides/models/slide_slide.py

```diff
@@ -562,3 +587,56 @@ def _default_website_meta(self):
         res['default_opengraph']['og:description'] = res['default_twitter']['twitter:description'] = self.description
         res['default_opengraph']['og:image'] = res['default_twitter']['twitter:image'] = "/web/image/slide.slide/%s/image_thumb" % (self.id)
         return res
+
+    def get_user_relation(self, user):
+        relation = self.user_relation_ids.search([('user_id', '=', user)])
+        if relation:
+            return relation.state
+
+    @api.model
+    def reorder_slides(self, slides_data=None):
+        ids = [e['id'] for e in slides_data]
+        slides_to_change = self.browse(ids)
+        for slide in slides_data:
+            s = slides_to_change.browse(slide['id'])
+            s.sudo().write({
+                'sequence': slide['sequence'],
+                'category_id': slide['category_id']
+            })
+
+
+class SlideQuestion(models.Model):
+    _name = "slide.question"
+    _rec_name = "question"
+
+    sequence = fields.Integer(default=10)
+    question = fields.Char(required=True, string="Question Name")
+    slide_id = fields.Many2one('slide.slide', string="Slide")
+    answer_ids = fields.One2many('slide.answer', 'question_id', string="Answer")
+
+    @api.constrains('answer_ids')
+    def _check_only_one_good_answer_and_at_least_two_possible_answers(self):
+        for question in self:
+            good_answer_count = 0
+            if len(question.answer_ids) < 2:
+                raise models.ValidationError('A question must have at least two possible answers') 
+            for answer in question.answer_ids:
+                if answer.is_correct:
+                    good_answer_count += 1
+                    if good_answer_count > 1:
+                        raise models.ValidationError('A question can only have one good answer')
+            if good_answer_count == 0:
+                raise models.ValidationError('A question must have a good answer')
+
+class SlideAnswer(models.Model):
+    _name = "slide.answer"
+    _rec_name = "text_value"
+
+    question_id = fields.Many2one('slide.question', string="Question")
+    text_value = fields.Char(required=True, string="Answer")
```

Comment:
Should probably be `fields.Char("Answer", required=True)`

Same for is_correct

Reply:
Done